src/Dependencies/PooledObjects/ArrayBuilder.cs

```diff
@@ -449,6 +449,13 @@ public static ArrayBuilder<T> GetInstance(int capacity, T fillWithValue)
             return builder;
         }
 
+        public static ImmutableArray<T> ToImmutable(ReadOnlySpan<T> items)
+        {
+            var builder = GetInstance(items.Length);
+            builder.AddRange(items);
+            return builder.ToImmutableAndFree();
+        }
```

Comment:
Why not just use `ImmutableArray.Create<T>(ReadOnlySpan<T>)`?

Reply:
agreed.  that's the existing fast-path.

Reply:
I mistakenly thought that path caused a builder allocation to occur. Looking now it's clear that doesn't happen. Will move to the suggested pattern.